mutable_tree.go

```diff
@@ -498,29 +541,57 @@ func (tree *MutableTree) SaveVersion() ([]byte, int64, error) {
 		return nil, version, fmt.Errorf("version %d was already saved to different hash %X (existing hash %X)", version, newHash, existingHash)
 	}
 
+	var batches []tmdb.Batch
+	var rootBatch tmdb.Batch
 	if tree.root == nil {
 		// There can still be orphans, for example if the root is the node being
 		// removed.
 		debug("SAVE EMPTY TREE %v\n", version)
-		tree.ndb.SaveOrphans(version, tree.orphans)
-		if err := tree.ndb.SaveEmptyRoot(version); err != nil {
+		batch := tree.ndb.saveOrphansEx(version, tree.orphans)
+		batches = append(batches, batch)
+		rootBatch = tree.ndb.db.NewBatch()
+		if err := tree.ndb.saveRootEx(rootBatch, []byte{}, version); err != nil {
 			return nil, 0, err
 		}
+		updatedNodesKeep.update(version, 0)
 	} else {
 		debug("SAVE TREE %v\n", version)
-		tree.ndb.SaveBranch(tree.root)
-		tree.ndb.SaveOrphans(version, tree.orphans)
-		if err := tree.ndb.SaveRoot(tree.root, version); err != nil {
+		var wg sync.WaitGroup
+		wg.Add(2)
+
+		var batch tmdb.Batch
+		go func() {
+			var count int64
+			_, count, batches = tree.ndb.SaveBranchEx(tree.root, 7)
+			wg.Done()
+			updatedNodesKeep.update(version, count*2)
+		}()
+		go func() {
+			batch = tree.ndb.saveOrphansEx(version, tree.orphans)
+			wg.Done()
+		}()
+		wg.Wait()
+
+		if batch != nil {
+			batches = append(batches, batch)
+		}
+		rootBatch = tree.ndb.db.NewBatch()
+		if err := tree.ndb.saveRootEx(rootBatch, tree.root.hash, version); err != nil {
 			return nil, 0, err
 		}
 	}
 
-	if err := tree.ndb.Commit(); err != nil {
-		return nil, version, err
-	}
+	func() {
```

Comment:
It seems to be an unnecessary function definition.
same comment with https://github.com/line/iavl/pull/30

Reply:
It's a place-holder to make it asynchronous function later.